### PR TITLE
[image_picker_platform_interface] Added pickMultiImage

### PR DESCRIPTION
**Pull request by danielroek**
Adds a pickMultiImnage method to the image_picker_platform_interface to prepare for Implementation in Android and iOS

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart [hunk 1]
```diff
@@ -36,6 +36,54 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
...
+
+  Future<List<dynamic>?> _pickMultiImagePath({
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+  }) {
+    if (imageQuality != null && (imageQuality < 0 || imageQuality > 100)) {
+      throw ArgumentError.value(
+          imageQuality, 'imageQuality', 'must be between 0 and 100');
+    }
```
Comment (mvanbeusekom) [inline packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart:67]: It looks like this check is not tested, would be good to also add a quick using test making sure te boundries work.

File: packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart [hunk 2]
```diff
@@ -36,6 +36,54 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
     return path != null ? PickedFile(path) : null;
   }
 
+  @override
+  Future<List<PickedFile>?> pickMultiImage({
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+  }) async {
+    List<dynamic>? paths = await _pickMultiImagePath(
```
Comment (mvanbeusekom) [inline packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart:45]: nit: could be marked `final`:

```suggestion
    final List<dynamic>? paths = await _pickMultiImagePath(
```

File: packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart [hunk 3]
```diff
@@ -36,6 +36,54 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
...
+      maxWidth: maxWidth,
+      maxHeight: maxHeight,
+      imageQuality: imageQuality,
+    );
+    if (paths == null) return null;
+
+    List<PickedFile> files = [];
```
Comment (mvanbeusekom) [inline packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart:52]: nit: could be marked `final`:

```suggestion
    final List<PickedFile> files = [];
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Look good to me, just one test missing and 2 small nits on local fields that could be marked `final`.

File: packages/image_picker/image_picker_platform_interface/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
+
+* Added pickMultiImage method
```
Comment (cyanglaz) [inline packages/image_picker/image_picker_platform_interface/CHANGELOG.md:3]:
```suggestion
* Add `pickMultiImage` method.
```

**Review by cyanglaz — APPROVED** [review]
LGTM % nits

**Review by mvanbeusekom — APPROVED** [review]
LGTM! Thanks @danielroek

**Pull request merged**